internals/cli/env_template.go

```diff
@@ -0,0 +1,389 @@
+
+func hasManagementSubCommands(cmd *cobra.Command) bool {
+	return len(managementSubCommands(cmd)) > 0
+}
+
+func operationSubCommands(cmd *cobra.Command) []*cobra.Command {
+	var cmds []*cobra.Command
+	for _, sub := range cmd.Commands() {
+		if sub.IsAvailableCommand() && !sub.HasSubCommands() {
+			cmds = append(cmds, sub)
+		}
+	}
+	return cmds
+}
+
+func managementSubCommands(cmd *cobra.Command) []*cobra.Command {
+	var cmds []*cobra.Command
+	for _, sub := range (*cmd).Commands() {
+		if sub.IsAvailableCommand() && sub.HasSubCommands() {
+			cmds = append(cmds, sub)
+		}
+	}
+	return cmds
+}
+
+func decoratedName(cmd cobra.Command) string {
+	return cmd.Name() + " "
+}
+
+func hasArgs(args []Argument) bool {
+	return len(args) > 0
+}
+
+func argUsages(args []Argument) string {
+	buf := new(bytes.Buffer)
+	lines := make([]string, 0, len(args))
+	maxlen := 0
+
+	cols := 80
+	if w, _, err := term.GetSize(0); err == nil {
+		cols = w
+	}
+
+	for _, arg := range args {
+		if arg.Hidden {
+			continue
+		}
+		line := "  "
+		if arg.Placeholder != "" {
+			line += arg.Placeholder
+		} else {
+			line += "<" + arg.Name + ">"
+		}
+
+		// This special character will be replaced with spacing once the
+		// correct alignment is calculated
+		line += "\x00"
+		if len(line) > maxlen {
+			maxlen = len(line)
+		}
+
+		line += arg.Description
+		lines = append(lines, line)
+	}
+
+	for _, line := range lines {
+		sidx := strings.Index(line, "\x00")
+		spacing := strings.Repeat(" ", maxlen-sidx)
+		// maxlen + 2 comes from + 1 for the \x00 and + 1 for the (deliberate) off-by-one in maxlen-sidx
+		fmt.Fprintln(buf, line[:sidx], spacing, wrap(maxlen+2, cols, line[sidx+1:]))
+	}
+
+	return buf.String()
+}
+
+func useLine(c *cobra.Command, args []Argument) string {
+	var useLine string
+
+	if c.HasParent() {
+		useLine = c.Parent().CommandPath() + " " + c.Use
+	} else {
+		useLine = c.Use
+	}
+
+	if c.HasAvailableFlags() && !strings.Contains(useLine, "[flags]") {
+		useLine += " [flags]"
+	}
+
+	for _, arg := range args {
+		if arg.Hidden {
+			continue
+		}
+		if arg.Placeholder != "" {
+			useLine += " " + arg.Placeholder
+		} else {
+			useLine += " <" + arg.Name + ">"
+		}
+	}
+
+	return useLine
+}
+
+func flagUsages(c CommandClause, isGlobal bool) string {
+	flagSet := c.Cmd.LocalFlags()
+	if isGlobal {
+		flagSet = c.Cmd.InheritedFlags()
+	}
+	buf := new(bytes.Buffer)
+
+	lines := make([]string, 0, flagSet.NFlag())
+
+	maxlen := 0
+
+	cols := 80
+	if w, _, err := term.GetSize(0); err == nil {
+		cols = w
+	}
+
+	flagSet.VisitAll(func(f *pflag.Flag) {
+		if f.Hidden {
+			return
+		}
+
+		line := ""
+		if f.Shorthand != "" && f.ShorthandDeprecated == "" {
+			line = fmt.Sprintf("  -%s, --%s", f.Shorthand, f.Name)
+		} else {
+			line = fmt.Sprintf("      --%s", f.Name)
+		}
+
+		// TODO: Decide whether we want to put the type as well. If not, remove lines.
+		//varname, _ := pflag.UnquoteUsage(f)
+		//if varname != "" {
+		//	line += " " + varname
+		//}
```

Comment:
This is indeed placed for a design decision.
Do we want to also display the type the flag is. E.g. for `acl ls`, flag `depth`, do we want to display `--depth string` or just the name of it?

That commented code adds the type: `string`, `int`, `duration`, as well as custom types made by us. 
The risk of adding the type is that the help text becomes crowded, and with custom types it can get confusing. 

I kept it commented just to know that we have this possibility as well in the template.
Feel free to give your opinion about whether it is a good idea to put the type in the help text as well or not.

Reply:
We can leave it out for now and add it later if we see a reason for it.

Reply:
Removed the commented code.